File: src/IsbInterop/Tasks/Proxies/Job.vb
Imports IsbInterop.Base.Proxies

Namespace Tasks.Proxies

  ''' <summary>
  ''' Обертка над IJob.
  ''' </summary>
  Friend Class Job
    Inherits CustomJob(Of IJobInfo)
    Implements IJob

    ''' <summary>
    ''' Получает информацию об объекте.
    ''' </summary>
    ''' <returns>Информация об объекте.</returns>
    Public Overrides Function GetInfo() As IJobInfo
      Dim rcwIJobInfo = GetRcwObjectInfo()
      Return New JobInfo(rcwIJobInfo, Scope)
    End Function

    ''' <summary>
    ''' Конструктор.
    ''' </summary>
    ''' <param name="rcwIJob">СOM-объект IJob.</param>
    ''' <param name="scope">Область видимости.</param>
    Public Sub New(rcwIJob As Object, scope As IScope)
      MyBase.New(rcwIJob, scope)
    End Sub
  End Class
End Namespace

File: src/IsbInterop/Tasks/Proxies/Job.cs
﻿using IsbInterop.Base.Proxies;
using IsbInterop.Edms.Proxies;

namespace IsbInterop.Tasks.Proxies
{
  /// <summary>
  /// Обертка над IJob.
  /// </summary>
  internal class Job : CustomJob<IJobInfo>, IJob
  {
    /// <summary>
    /// Получает информацию об объекте.
    /// </summary>
    /// <returns>Информация об объекте.</returns>
    public override IJobInfo GetInfo()
    {
      var rcwIJobInfo = GetRcwObjectInfo();
      return new JobInfo(rcwIJobInfo, Scope);
    }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwIJob">СOM-объект IJob.</param>
    /// <param name="scope">Область видимости.</param>
    public Job(object rcwIJob, IScope scope) : base(rcwIJob, scope) { }
  }
}
